### PR TITLE
WIP Edge case where schema diffing has NPE

### PR DESCRIPTION
**Pull request by dondonz**
WIP, this is a failing test for now

There are a number of schema changes, this particular combination is causing an NPE because Mappings is null

### CODE REVIEW

File: src/test/groovy/graphql/schema/diffing/ana/EditOperationAnalyzerAppliedDirectivesTest.groovy [hunk 1]
```diff
@@ -32,6 +32,1183 @@ import static graphql.schema.diffing.ana.SchemaDifference.UnionModification
 
 class EditOperationAnalyzerAppliedDirectivesTest extends Specification {
```
Comment (dondonz) [inline src/test/groovy/graphql/schema/diffing/ana/EditOperationAnalyzerAppliedDirectivesTest.groovy:33]: PS this is a quick reproduction, I will move the test to another file later

**Comment by dondonz** [comment]
Fixed by #3304
